here-naksha-app-service/src/test/java/com/here/naksha/app/service/IterateFeaturesTest.java

```diff
@@ -180,31 +181,50 @@ void tc1104_testIterateInTwoPagesWithPropSelection() throws URISyntaxException,
     final String firstStreamId = UUID.randomUUID().toString();
-    final String firstExpectedBodyPart = loadFileOrFail("ReadFeatures/Iterate/TC1104_testIterateInTwoPagesWithPropSelection/iterate_response_1.json")
-            .replaceAll("\\{\\{streamId}}",firstStreamId);
+    final String firstExpectedBodyPart = loadFileOrFail(
+        "ReadFeatures/Iterate/TC1104_testIterateInTwoPagesWithPropSelection/iterate_response_1.json")
+        .replaceAll("\\{\\{streamId}}", firstStreamId);
     final String selectionParams = "selection=p.speedLimit,p.unknown_prop,%s".formatted(urlEncoded("p.@ns:com:here:xyz.tags"));
 
     // When: First iterate Features request is submitted to NakshaHub
-    HttpResponse<String> response = nakshaClient.get("hub/spaces/" + SPACE_ID + "/iterate" + "?" + limitQueryParam + "&" + selectionParams, firstStreamId);
+    HttpResponse<String> response = nakshaClient.get("hub/spaces/" + SPACE_ID + "/iterate" + "?" + limitQueryParam + "&" + selectionParams,
+        firstStreamId);
     // Then: Perform assertions
     ResponseAssertions.assertThat(response)
-            .hasStatus(200)
-            .hasStreamIdHeader(firstStreamId)
-            .hasJsonBody(firstExpectedBodyPart, "First Iterate response body doesn't match", true);
+        .hasStatus(200)
+        .hasStreamIdHeader(firstStreamId)
+        .hasJsonBody(firstExpectedBodyPart, "First Iterate response body doesn't match", true);
 
     // Given: iterate parameters for second request
     final String handleQueryParam = "handle=" + urlEncoded(parseJson(response.body(), XyzFeatureCollection.class).getNextPageToken());
-    final String secondExpectedBodyPart = loadFileOrFail("ReadFeatures/Iterate/TC1104_testIterateInTwoPagesWithPropSelection/iterate_response_2.json");
+    final String secondExpectedBodyPart = loadFileOrFail(
+        "ReadFeatures/Iterate/TC1104_testIterateInTwoPagesWithPropSelection/iterate_response_2.json");
     final String secondStreamId = UUID.randomUUID().toString();
     final String selectionWrongDelimiterParams = "selection=p.speedLimit+p.unknown_prop";
 
     // When: Second iterate Features request is submitted to NakshaHub
-    response = nakshaClient.get("hub/spaces/" + SPACE_ID + "/iterate" + "?" + handleQueryParam + "&" + selectionWrongDelimiterParams, secondStreamId);
+    response = nakshaClient.get("hub/spaces/" + SPACE_ID + "/iterate" + "?" + handleQueryParam + "&" + selectionWrongDelimiterParams,
+        secondStreamId);
     // Then: Perform assertions
     ResponseAssertions.assertThat(response)
-            .hasStatus(400)
-            .hasStreamIdHeader(secondStreamId)
-            .hasJsonBody(secondExpectedBodyPart, "Final Iterate response body doesn't match");
+        .hasStatus(400)
+        .hasStreamIdHeader(secondStreamId)
+        .hasJsonBody(secondExpectedBodyPart, "Final Iterate response body doesn't match");
   }
 
+  @Test
+  void tc1105_iterateShouldRespectAndPropagateLimit() throws Exception {
```

Comment:
tc1101 and tc1102 validates multiple iterate operations, validating each page fetch. So this test seems duplicate of what is already covered and can be safely removed. Technically no harm having it though, just that it can cause confusion in understanding what is the difference.

Reply:
I agree - this test covers logic that was tested already. I used this one mainly for quick debug sessions - will remove